Build failure: pkgsLLVM.alsa-firmware

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.alsa-firmware`

### Build log

```
Running phase: configurePhase
calling 'preConfigure' function hook '_multioutConfig'
fixing libtool script ./ltmain.sh
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/mmhpq6dhbhs5a3nwk4qw80varw8xzrwl-bash-5.2p26/bin/sh"
configure flags: --disable-dependency-tracking --prefix=/nix/store/9g72gw3kbkcrqwydf5r7nd23rj66nng6-alsa-firmware-1.2.4 --with-hotplug-dir=\$\(out\)/lib/firmware --build=aarch64-unknown-linux-gnu --host=aarch64-unknown-linux-gnu
checking for aarch64-unknown-linux-gnu-gcc... aarch64-unknown-linux-gnu-gcc
checking whether the C compiler works... no
configure: error: in '/build/alsa-firmware-1.2.4':
configure: error: C compiler cannot create executables
See 'config.log' for more details
```

### Additional context

Discovered by https://github.com/RossComputerGuy/nixpkgs-llvm-ws/runs/28310158078

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@L-as

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/al/alsa-firmware/package.nix
+++ b/pkgs/by-name/al/alsa-firmware/package.nix
@@ -1,6 +1,13 @@
-{ lib, buildPackages, stdenvNoCC, autoreconfHook, fetchurl, fetchpatch }:
+{
+  lib,
+  stdenv,
+  buildPackages,
+  autoreconfHook,
+  fetchurl,
+  fetchpatch,
+}:
 
-stdenvNoCC.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "alsa-firmware";
   version = "1.2.4";
 
@@ -17,12 +24,13 @@ stdenvNoCC.mkDerivation rec {
     })
   ];
 
-  depsBuildBuild = [ buildPackages.stdenv.cc ];
   nativeBuildInputs = [ autoreconfHook ];
 
-  configureFlags = [
-    "--with-hotplug-dir=$(out)/lib/firmware"
-  ];
+  configureFlags = [ "--with-hotplug-dir=$(out)/lib/firmware" ];
+
+  depsBuildBuild = lib.optional (
+    stdenv.buildPlatform != stdenv.hostPlatform || stdenv.hostPlatform.isAarch64
+  ) buildPackages.stdenv.cc;
 
   dontStrip = true;
 

